profiling/src/profile/pprof/sliced_proto.rs

```diff
@@ -0,0 +1,153 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2023-Present Datadog, Inc.
+
+//! This file is a companion to the standard pprof protobuf definition.
+//!
+//! When `prost` serializes the standard `pprof` protobuf, it does a batch
+//! operation:
+//! 1. the entire `internal::Profile` is converted to `pprof::Profile`,
+//! 2. which is then serialized into a byte-vector holding the protobuf,
+//! 3. which is then compressed.
```

Comment:
Minor: Technically, 1. is on our side, not prost that does it, and 3. prost does do at all